src/contentmanager.cpp

```diff
@@ -420,17 +433,40 @@ ContentManager::BookInfo ContentManager::getBookInfos(QString id, const QStringL
     return values;
 }
 
-void ContentManager::openBookWithIndex(const QModelIndex &index)
+ContentManager::BookState ContentManager::getBookState(QString bookId)
 {
+    if ( const auto downloadState = m_downloads.value(bookId) ) {
+        return downloadState->paused
+             ? BookState::DOWNLOAD_PAUSED
+             : BookState::DOWNLOADING;
+             // TODO: a download may be in error state
+    }
+
     try {
-        auto bookNode = static_cast<Node*>(index.internalPointer());
-        const QString bookId = bookNode->getBookId();
-        // throws std::out_of_range if the book isn't available in local library
-        const kiwix::Book& book = mp_library->getBookById(bookId);
-        if ( !book.getDownloadId().empty() )
-            return;
+        const kiwix::Book& b = mp_library->getBookById(bookId);
+        return b.getDownloadId().empty()
```

Comment:
It happens for a short span of time (at most a few milliseconds) between the instant when `bookId` is removed from `m_downloads` and when `downloadId` is reset in the book. But this code was mainly written in order to not leave loopholes rather than due to that detail of the current implementation.